crates/edr_provider/src/lib.rs

```diff
@@ -514,3 +518,70 @@ fn to_json_with_traces<T: serde::Serialize, LoggerErrorT: Debug>(
         traces: value.1,
     })
 }
+
+// Rust port of https://github.com/NomicFoundation/hardhat/blob/024d72b09c6edefb00c012e9514a0948c255d0ab/v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/utils/convert-to-edr.ts#L176
+/// This normalization is done because this is the format Hardhat expects
+fn normalise_rpc_debug_trace<LoggerErrorT: Debug>(
```

Comment:
Please move this to the same file as the `handle_` function I mentioned in my other comment.